### ISSUE TITLE
Incorrect removal of "for foo; do" semicolon in case statement

### ISSUE THREAD

**Issue by ssmout** [mvdan/sh#281]
I have this minimal `example.sh`:
```sh
case $1 in
a) ;;
b)
        for arg; do
                echo $arg
        done
        ;;
esac
```

If I run shfmt:
```console
$ shfmt -version
v2.5.0
$ shfmt example.sh
```
```sh
case $1 in
a) ;;
b)
        for arg do
                echo $arg
        done
        ;;
esac
```

The semicolon between `arg` and `do` has been removed, which breaks the script.

The problem doesn't happen if the `for` loop is in the first `a)` case, or outside of a `case` statement. It does happen with a more interesting loop; e.g. `for arg in 1 2 3; do` and `for arg in "$@"; do` still result in the semicolon removal.

**Comment by mvdan** [comment]
Sorry that you've run into this regression! This has already been reported and fixed in #272, but I haven't done a release yet.

There's more that I want to do for 2.6.0, so I might release 2.5.1 with this fix and any other regression fixes. Until then, I believe 2.4.0 should work just fine in this case.

**Comment by ssmout** [comment]
You're absolutely right; it works at 2.4 and head. Sorry for my search fail.

**Comment by mvdan** [comment]
I've published 2.5.1 with a backport of that fix, and am building binaries to upload as I write this. Please give that a go; it should fix your problem.

**Comment by ssmout** [comment]
Yep, fixed. Thanks!
